src/ceph-daemon

```diff
@@ -0,0 +1,1289 @@
+        makedirs(log_dir)
+    if args.name:
+        if '.' in args.name:
+            (daemon_type, daemon_id) = args.name.split('.')
+        else:
+            daemon_type = args.name
+            daemon_id = None
+    else:
+        daemon_type = 'osd'  # get the most mounts
+        daemon_id = None
+    mounts = get_container_mounts(args.fsid, daemon_type, daemon_id)
+    if args.config:
+        mounts[pathify(args.config)] = '/etc/ceph/ceph.conf:z'
+    if args.keyring:
+        mounts[pathify(args.keyring)] = '/etc/ceph/ceph.keyring:z'
+    c = CephContainer(
+        image=args.image,
+        entrypoint='doesnotmatter',
+        args=[],
+        podman_args=['--privileged'],
+        volume_mounts=mounts)
+    subprocess.call(c.shell_cmd())
+
+##################################
+
+def command_enter():
+    (daemon_type, daemon_id) = args.name.split('.')
+    c = get_container(args.fsid, daemon_type, daemon_id)
+    subprocess.call(c.exec_cmd(['bash']))
+
+##################################
+
+def command_exec():
+    (daemon_type, daemon_id) = args.name.split('.')
+    c = get_container(args.fsid, daemon_type, daemon_id,
+                      privileged=args.privileged)
+    subprocess.call(c.exec_cmd(args.command))
+
+##################################
+
+def command_ceph_volume():
+    log_dir = get_log_dir(args.log_dir, args.fsid)
+    makedirs(log_dir)
+
+    mounts = get_container_mounts(args.fsid, 'osd', None)
+
+    tmp_config = None
+    tmp_keyring = None
+
+    if args.config_and_keyring:
+        # note: this will always pull from args.config_and_keyring (we
+        # require it) and never args.config or args.keyring.
+        (config, keyring) = get_config_and_keyring()
+
+        # tmp keyring file
+        tmp_keyring = tempfile.NamedTemporaryFile(mode='w')
+        os.fchmod(tmp_keyring.fileno(), 0o600)
+        tmp_keyring.write(keyring)
+        tmp_keyring.flush()
+
+        # tmp config file
+        tmp_config = tempfile.NamedTemporaryFile(mode='w')
+        os.fchmod(tmp_config.fileno(), 0o600)
+        tmp_config.write(config)
+        tmp_config.flush()
+
+        mounts[tmp_config.name] = '/etc/ceph/ceph.conf:z'
+        mounts[tmp_keyring.name] = '/var/lib/ceph/bootstrap-osd/ceph.keyring:z'
+
+    c = CephContainer(
+        image=args.image,
+        entrypoint='/usr/sbin/ceph-volume',
+        args=args.command,
+        podman_args=['--privileged'],
+        volume_mounts=mounts,
+    )
+    subprocess.call(c.run_cmd())
+
+##################################
+
+def command_unit():
+    (daemon_type, daemon_id) = args.name.split('.')
+    c = get_container(args.fsid, daemon_type, daemon_id)
+    unit_name = get_unit_name(args.fsid, daemon_type, daemon_id)
+    subprocess.call([
+        'systemctl',
+        args.command,
+        unit_name])
+
+##################################
+
+def command_ls():
+    ls = []
+
+    # /var/lib/ceph
+    if os.path.exists(args.data_dir):
```

Comment:
this `if` condition is too deep... one way of reducing the nesting is by negating and returning:

```suggestion
    if not os.path.exists(args.data_dir):
        return
```